### DISCUSSION TITLE
Host Caddy on separate server/Avoid IP block

### DISCUSSION THREAD

**Discussion by rayegun** [coder/code-server#2109] [Q&A] [answered]
I have a somewhat noob question. I have two subdomains on my site, one for vscode, and one for another service. That other service is hosted on digital ocean, and is available anywhere. My code-server instance is hosted on a home server. Unfortunately it is not available anywhere, for some reason the IP is blocked (at work for instance). I'm unsure why my home IP would be blocked, but it is. 

My question is whether there's a way to host the code server itself at home, but expose it through Digital ocean. The DNS is currently handled by them already.

**Comment by code-asher** [comment, marked as answer]
Sounds like you'll need to proxy to your home server from Digital Ocean. Maybe you could point Caddy on Digital Ocean to your home server address? Or set up a VPN on the droplet that connects to your home server. I haven't done anything like this so I don't have any super specific advice unfortunately.

We're also working on a cloud feature that'll make it easy to access your home server without having to access the IP directly. https://github.com/cdr/code-server/discussions/2077
